### ISSUE TITLE
False positive F821: string Literal in combintation with future import

### ISSUE THREAD

**Issue by twoertwein** [astral-sh/ruff#6030]
I believe this was not triggered with 0.0.278 but is triggered with 0.0.280
```py
from __future__ import annotations  # without this line ruff does not report F821

from typing import Literal, Final

from typing_extensions import assert_type

CONSTANT: Final = "ns"

assert_type(CONSTANT, Literal["ns"])
```
```sh
ruff --select F821 test.py 
test.py:9:32: F821 Undefined name `ns`
```

**Comment by charliermarsh** [comment]
Thanks, I'll take a look.
